curve-exchange/src/addressProvider.ts

```diff
@@ -0,0 +1,53 @@
+import { BigInt } from "@graphprotocol/graph-ts";
+import { AddressModified } from "../generated/CurveExchangeAddressProvider/AddressProvider";
+
+import { PoolRegistry, AddressProvider, MetaPoolFactory } from "../generated/schema";
+import {
+  PoolRegistry as PoolRegistryTemplate,
+  MetaPoolFactory as MetaPoolFactoryTemplate,
+} from "../generated/templates";
+
+const REGISTRY_ID = 0;
+const METAPOOL_FACTORY_ID = 3;
+
+export function handleAddressModified(event: AddressModified): void {
+  let id = event.params.id;
+  let newAddress = event.params.new_address;
+
+  // create address provider entity
+  let addressProvider = AddressProvider.load(event.address.toHexString());
+
+  if (addressProvider == null) {
+    addressProvider = new AddressProvider(event.address.toHexString());
+    addressProvider.save();
+  }
+
+  // create registry entity
+  if (id == BigInt.fromI32(REGISTRY_ID)) {
+    let poolRegistry = PoolRegistry.load(newAddress.toHexString());
+    if (poolRegistry == null) {
+      poolRegistry = new PoolRegistry(newAddress.toHexString());
+      poolRegistry.save();
+
+      // don't use this address as that registry is not functioning
+      if (poolRegistry.id != "0xe2470c5e330a34d706f93d50658ba52d18512f7a") {
```

Comment:
I think it would be safer to add .toLowerCase() in case of any future update converts addresses to it's checksum compliant version (case-sensitive)

Reply:
Actually best if we save it as lowercase instead of actually only doing it for this comparison

Reply:
Sorry saw the comments only now, done!